addons/resource/models/resource.py

```diff
@@ -622,7 +622,9 @@ def interval_dt(interval):
 
         if not dt.tzinfo or search_range and not (search_range[0].tzinfo and search_range[1].tzinfo):
             raise ValueError('Provided datetimes needs to be timezoned')
-        dt = dt.astimezone(timezone(self.tz))
+
+        tz = resource.tz if resource else self.tz
```

Comment:
Hello !

Thanks for the feedback :)

How did you get that issue ? I see that each time the method is called, it is passing the same resource than the one in self: https://github.com/odoo-dev/odoo/blame/15.0-thirteen-month-yti/addons/resource/models/resource.py#L954-L964

Reply:
Hello @tivisse,
Actually self is the resource.calendar and the resource is passed in (thank you @tlibioulle for your insight).